### PR TITLE
check vsc null pointer

### PR DESCRIPTION
**Pull request by lilongfeng0902**
Thank you for contributing to Velero!

# Please add a summary of your change
Fix a little problem, add a null pointer to judge.
# Does your change fix a particular issue?
none
Fixes #(issue)
none
# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

Signed-off-by: 李龙峰 [lilongfeng@inspur.com](mailto:lilongfeng@inspur.com)

### CODE REVIEW

**Comment by sseago** [comment]
The changelog filename doesn't match the PR, since a new PR was created. You need to rename "changelogs/unreleased/5213-lilongfeng0902" to "changelogs/unreleased/5217-lilongfeng0902" in this PR.

**Comment by lilongfeng0902** [comment]
> The changelog filename doesn't match the PR, since a new PR was created. You need to rename "changelogs/unreleased/5213-lilongfeng0902" to "changelogs/unreleased/5217-lilongfeng0902" in this PR.

Thanks for your review, I have did it.

File: pkg/controller/backup_controller.go [hunk 1]
```diff
@@ -951,7 +951,7 @@ func (c *backupController) deleteVolumeSnapshot(volumeSnapshots []*snapshotv1api
 			if vs.Status.BoundVolumeSnapshotContentName != nil &&
 				len(*vs.Status.BoundVolumeSnapshotContentName) > 0 {
 				vsc = vscMap[*vs.Status.BoundVolumeSnapshotContentName]
```
Comment (blackpiglet) [inline pkg/controller/backup_controller.go:953]: Thanks for contribution.
This is useful prorection checking, but I think if vsc is nil, or cannot be found from the vscMap, we should record an error, then return.
In normal case, this shouldn't happen, but if this happens, it means VSC is deleleted accidently, and the external snapshot controller should recreate vsc sometime later, in this case, I think no need to delete vs, just leaving with error message should be enough.

Comment (lilongfeng0902) [reply pkg/controller/backup_controller.go:952]: Have processed as what you suggest, please review again. Thanks!

**Comment by blackpiglet** [comment]
Thanks for the effort.
Now I only have one comment left. Please squash the commit into one, and force push to make the commit list simple and clean.

File: pkg/controller/backup_controller.go [hunk 2]
```diff
@@ -28,7 +28,6 @@ import (
 	"sync"
 	"time"
 
-	"github.com/apex/log"
```
Comment (blackpiglet) [inline pkg/controller/backup_controller.go:31]: Please put this line back. There is still some code using this log library.

Comment (lilongfeng0902) [reply pkg/controller/backup_controller.go]: I am so sorry for the mistake. Please review again.

**Comment by lilongfeng0902** [comment]
> Thanks for the effort. Please squash the commit into one, and force push to make the commit list simple and clean.

I squash the commit into one, and force push the commit. Thanks.

File: pkg/controller/backup_controller.go [hunk 3]
```diff
@@ -951,8 +951,11 @@ func (c *backupController) deleteVolumeSnapshot(volumeSnapshots []*snapshotv1api
 			if vs.Status.BoundVolumeSnapshotContentName != nil &&
 				len(*vs.Status.BoundVolumeSnapshotContentName) > 0 {
 				vsc = vscMap[*vs.Status.BoundVolumeSnapshotContentName]
-				if vsc.Spec.DeletionPolicy == snapshotv1api.VolumeSnapshotContentDelete {
+				if nil != vsc && vsc.Spec.DeletionPolicy == snapshotv1api.VolumeSnapshotContentDelete {
```
Comment (blackpiglet) [inline pkg/controller/backup_controller.go:954]:
``` go
				if nil == vsc {
					logger.Errorf("Not find %s from the vscMap", vs.Status.BoundVolumeSnapshotContentName)
					return
				}
				if vsc.Spec.DeletionPolicy == snapshotv1api.VolumeSnapshotContentDelete {
					modifyVSCFlag = true
				}
```

Comment (lilongfeng0902) [reply pkg/controller/backup_controller.go]: > Sorry, I didn't notice this in last review. It should be something like this.
> 
> ```go
> 				if nil == vsc {
> 					logger.Errorf("Not find %s from the vscMap", vs.Status.BoundVolumeSnapshotContentName)
> 					return
> 				}
> 				if vsc.Spec.DeletionPolicy == snapshotv1api.VolumeSnapshotContentDelete {
> 					modifyVSCFlag = true
> 				}
> ```

Yes，that's look like good, I have did that. Please check again. Thank you.

**Review by blackpiglet — APPROVED** [review]

File: pkg/controller/backup_controller.go [hunk 4]
```diff
@@ -951,6 +951,10 @@ func (c *backupController) deleteVolumeSnapshot(volumeSnapshots []*snapshotv1api
 			if vs.Status.BoundVolumeSnapshotContentName != nil &&
 				len(*vs.Status.BoundVolumeSnapshotContentName) > 0 {
 				vsc = vscMap[*vs.Status.BoundVolumeSnapshotContentName]
+				if nil != vsc {
```
Comment (reasonerjt) [inline pkg/controller/backup_controller.go:954]: Should  it be `nil== vsc`?

Comment (lilongfeng0902) [reply pkg/controller/backup_controller.go]: Yes, you're right. Too careless……

**Review by reasonerjt — APPROVED** [review]

**Review by blackpiglet — APPROVED** [review]

**Pull request merged**